Update HTTP integration docs for UI configuration (#45302)
home-assistant/home-assistant.github.io: source/_integrations/http.markdown

  - '@home-assistant/core'
ha_domain: http
ha_integration_type: system
related:
  - docs: /docs/configuration/remote/
    title: Configuring remote access
---

The **HTTP** {% term integration %} serves the Home Assistant frontend and powers the [REST API](https://developers.home-assistant.io/docs/api/rest/), the [Python API](https://developers.home-assistant.io/docs/api_lib_index/), and the [WebSocket API](https://developers.home-assistant.io/docs/api/websocket/). It is part of every Home Assistant installation and does not need to be added manually.

## Configuration

> The HTTP integration is set up automatically. To change the listening port, set up SSL, allow CORS origins, configure trusted reverse proxies, or tune IP banning, go to {% my network title="**Settings** > **System** > **Network**" %} and edit the **HTTP server** section. Changes take effect after a Home Assistant restart.

Reviewer: Let's use numbered steps for the procedure:

```suggestion
The HTTP integration is set up automatically.

To change the listening port, set up SSL, allow CORS origins, configure trusted reverse proxies, or tune IP banning:

1. Go to {% my network title="**Settings** > **System** > **Network**" %}.
2. Edit the **HTTP server** section.
3. Restart Home Assistant.
```

Enable IP banning:
  description: "<a id='ip_ban_enabled'></a>Automatically ban remote clients after repeated failed login attempts. Enabled by default."
Login attempts before ban:
  description: "<a id='login_attempts_threshold'></a>Failed login attempts from a single IP address before that address is banned, when **Enable IP banning** is on. Set to `-1` to disable adding new bans automatically. The default is `-1`."
{% endconfiguration_basic %}

## Migrating from YAML

Before Home Assistant 2026.6, the HTTP integration was configured in {% term "`configuration.yaml`" %}. On the first startup after the upgrade, your existing `http:` block is imported into the UI and a repair issue is raised under {% my repairs title="**Settings** > **System** > **Repairs**" %} prompting you to remove the YAML configuration.

To complete the migration:

1. Go to {% my network title="**Settings** > **System** > **Network**" %} and verify that the **HTTP server** section shows your imported values.
2. Remove the `http:` block from {% term "`configuration.yaml`" %}.

> 3. Restart Home Assistant.

Reviewer: Let's split the first step:

```suggestion
1. Go to {% my network title="**Settings** > **System** > **Network**" %}
2. Verify that the **HTTP server** section shows your imported values.
3. Remove the `http:` block from {% term "`configuration.yaml`" %}.
4. Restart Home Assistant.
```

}
```

For more examples, see the [HTTP binary sensor](#examples) section.

## Troubleshooting

### The HTTP server failed to start

If a stored value prevents the HTTP server from starting, for example an SSL certificate path that no longer exists, Home Assistant falls back to default HTTP settings on port `8123` without SSL so you can recover access. A repair issue titled **HTTP configuration was rejected on startup** is raised under {% my repairs title="**Settings** > **System** > **Repairs**" %} with the underlying error. Open the **HTTP server** section in {% my network title="**Settings** > **System** > **Network**" %}, correct the values, save, and restart.

### Repair issue: the HTTP YAML configuration is deprecated

> This issue appears when an `http:` block is still present in {% term "`configuration.yaml`" %} after the values were imported into the UI. Verify your settings under {% my network title="**Settings** > **System** > **Network**" %}, remove the `http:` block, and restart Home Assistant.

Reviewer: Let's break the paragraphs into problem-solution:

```suggestion
If a stored value prevents the HTTP server from starting, for example an SSL certificate path that no longer exists, Home Assistant falls back to default HTTP settings on port `8123` without SSL so you can recover access. A repair issue titled **HTTP configuration was rejected on startup** is raised under {% my repairs title="**Settings** > **System** > **Repairs**" %} with the underlying error.

To fix this, open the **HTTP server** section in {% my network title="**Settings** > **System** > **Network**" %}, correct the values, save, and restart.

### Repair issue: the HTTP YAML configuration is deprecated

This issue appears when an `http:` block is still present in {% term "`configuration.yaml`" %} after the values were imported into the UI.

To fix this, verify your settings under {% my network title="**Settings** > **System** > **Network**" %}, remove the `http:` block, and restart Home Assistant.
```
